lib/glob.js

```diff
@@ -1,12 +1,19 @@
 'use strict';
 
-var braces = require('braces');
-var brackets = require('expand-brackets');
-var extglob = require('extglob');
-var parse = require('parse-glob');
+var isExtglob = require('is-extglob');
 var chars = require('./chars');
 
 /**
+ * Lazily required module dependencies.
+ */
+
+var lazy = require('lazy-cache')(require);
+var lazyBraces = lazy('braces');
```

Comment:
> you have to call the returned function to actually initialize the module... I had assumed it was like a seamless wrapper

yeah that's an ugly side-effect of using lazy-caching. I played around with a few different ideas to try to hide or move the function call elsewhere, but the complication didn't seem worth it. 

fwiw I started using the `lazy*` naming because I have had the same response you had _looking at my own code_ after implementing lazy caching lol - anyway it seems more clear why the function is being called this way.